app/workers/document_list_export_worker.rb

```diff
@@ -15,8 +15,13 @@ def perform(filter_options, user_id)
 
   def send_mail(csv_filename, user, filter)
     MailNotifications.document_list(csv_filename, user.email, filter.page_title).deliver_now
-  rescue Notifications::Client::BadRequestError
-    # unrecoverable when running asynchronously
+  rescue Notifications::Client::BadRequestError => e
+    if !%w[integration staging].include?(ENV["SENTRY_CURRENT_ENV"]) || e.message !~ /team-only API key/
```

Comment:
Could you write some corresponding tests in [DocumentListExportWorkerTest](https://github.com/alphagov/whitehall/blob/4e2df36b2ba62a11986fa17fa5ce9983d8a25bc0/test%2Funit%2Fworkers%2Fdocument_list_export_worker_test.rb)?